Its.Log.Monitoring.UnitTests/AssertionTests.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Its.Log.Monitoring.UnitTests
+{
+    [TestFixture]
+    public class AssertionTests
+    {
+        private readonly StatefulResult<IEnumerable<int>, long> five = Enumerable.Range(1, 5).CountOf(_ => true);
+        private readonly IEnumerable<Result> results = Enumerable.Range(1, 10).Select(i => new Result(i % 2 == 0));
+
+        public class Result
+        {
+            public Result(bool success)
+            {
+                Success = success;
+            }
+
+            public bool Success { get; }
+        }
+
+        [Test]
+        public void BeGreaterThan_throws_when_expected()
+        {
+            Action lower = () => five.Should().BeGreaterThan(4);
+            lower.ShouldNotThrow();
+
+            Action equalTo = () => five.Should().BeGreaterThan(5);
+            equalTo.ShouldThrow<StatefulResultAssertionException<IEnumerable<int>>>();
+
+            Action upper = () => five.Should().BeGreaterThan(6);
+            upper.ShouldThrow<StatefulResultAssertionException<IEnumerable<int>>>();
+        }
+
+        [Test]
+        public void BeGreaterThanOrEqualTo_throws_when_expected()
+        {
+            Action lower = () => five.Should().BeGreaterThanOrEqualTo(4);
+            lower.ShouldNotThrow();
+
+            Action equalTo = () => five.Should().BeGreaterThanOrEqualTo(5);
+            equalTo.ShouldNotThrow();
+
+            Action upper = () => five.Should().BeGreaterThanOrEqualTo(6);
+            upper.ShouldThrow<StatefulResultAssertionException<IEnumerable<int>>>();
+        }
```

Comment:
Same. If you expand the test name to clarify "when expected" you'll get tests that actually describe the behavior we should be seeing.